# EDA
___

In [1]:
# imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats

from scipy.sparse import csr_matrix, save_npz, load_npz

In [2]:
df = pd.read_csv('../../Data/Movies/IMDB_Movies_cleaned_v01.csv')
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language                      director  \
0       153  Germany   German                    Fritz Lang   
1        87      USA  English               Charles Chaplin   
2        87      USA  English               Charles Chaplin   
3        83      USA  English  William Cottrell, David Hand   
4       238      USA  English  Victor Fleming, George Cukor   

                             writer               production_company  \
0  Thea von Harbou, Thea von Harbou             Universum Film (UFA)   
1                   Charles Chaplin      Charles Chaplin Productions   
2                   Charles Chaplin      Charles Chaplin Productions   
3        Jacob Grimm, Wilhelm Grimm          Walt Disney Productions   
4  Margaret Mitchell, Sidney Howard  Selznick International Pictures   

                                              actors  avg_vote   votes  \
0  Alfred Abel, Gustav Fröhlich, Rudolf Klein-Rog...       8.3  148396   
1  Virginia Cherrill, Florence Lee, Harry Myers, ...       8.5  152716   
2  Charles Chaplin, Paulette Goddard, Henry Bergm...       8.5  197969   
3  Roy Atwell, Stuart Buchanan, Adriana Caselotti...       7.6  168735   
4  Thomas Mitchell, Barbara O'Neil, Vivien Leigh,...       8.1  269664   

    budget  usa_gross_income  worldwide_gross_income  metascore  \
0   780000           1236166                 1349711       98.0   
1  1500000             19181                   32609       99.0   
2  1500000            163577                  445226       96.0   
3  1499000         184925486               184925486       95.0   
4  3977000         200852579               402352579       97.0   

   reviews_from_users  reviews_from_critics  
0               471.0                 194.0  
1               270.0                 120.0  
2               262.0                 146.0  
3               229.0                 166.0  
4               808.0                 183.0

In [3]:
df.describe()

year     duration     avg_vote         votes        budget  \
count  6459.000000  6459.000000  6459.000000  6.459000e+03  6.459000e+03   
mean   2003.381174   107.951231     6.444682  9.354221e+04  3.049212e+07   
std      11.587292    18.858011     0.986718  1.606785e+05  3.987928e+07   
min    1927.000000    63.000000     1.400000  1.000000e+02  2.000000e+01   
25%    1999.000000    95.000000     5.900000  1.141400e+04  5.749889e+06   
50%    2006.000000   105.000000     6.500000  3.749300e+04  1.600000e+07   
75%    2012.000000   118.000000     7.100000  1.036910e+05  3.800000e+07   
max    2019.000000   272.000000     9.300000  2.159628e+06  3.560000e+08   

       usa_gross_income  worldwide_gross_income    metascore  \
count      6.459000e+03            6.459000e+03  6459.000000   
mean       4.097111e+07            8.829865e+07    54.955411   
std        6.704754e+07            1.743236e+08    18.092700   
min        9.500000e+01            9.500000e+01     1.000000   
25%        1.548588e+06            4.631308e+06    42.000000   
50%        1.733637e+07            2.685746e+07    55.000000   
75%        5.156309e+07            9.209381e+07    68.000000   
max        9.366622e+08            2.797801e+09   100.000000   

       reviews_from_users  reviews_from_critics  
count         6459.000000           6459.000000  
mean           297.004800            147.188574  
std            451.897507            124.048523  
min              1.000000              1.000000  
25%             74.000000             58.000000  
50%            164.000000            113.000000  
75%            339.000000            197.000000  
max           8302.000000            987.000000

In [4]:
df.loc[df['budget'] <= 200]

title  year             genre  duration      country language  \
363   Poltergeist  1982  Horror, Thriller       114          USA  English   
6432        Grass  2018             Drama        66  South Korea   Korean   

           director                           writer  \
363     Tobe Hooper  Steven Spielberg, Michael Grais   
6432  Sang-soo Hong                    Sang-soo Hong   

             production_company  \
363   Metro-Goldwyn-Mayer (MGM)   
6432             Jeonwonsa Film   

                                                 actors  avg_vote   votes  \
363   Craig T. Nelson, JoBeth Williams, Beatrice Str...       7.3  132785   
6432  Min-hee Kim, Joobong Kee, Jae-hong Ahn, Jin-yo...       6.8     643   

      budget  usa_gross_income  worldwide_gross_income  metascore  \
363       20          77142388                77142388       79.0   
6432      82              8084                   53503       77.0   

      reviews_from_users  reviews_from_critics  
363                375.0                 250.0  
6432                 1.0                  30.0

In [5]:
# Budget for "Pooltergeist" is actually $10.7 (M) - https://www.imdb.com/title/tt0084516/?ref_=nv_sr_srsg_0
# Budget for "Grass" is technically correct - https://www.imdb.com/title/tt7913172/?ref_=fn_al_tt_1
df.loc[363, 'budget'] = 10_700_000

df.describe()

year     duration     avg_vote         votes        budget  \
count  6459.000000  6459.000000  6459.000000  6.459000e+03  6.459000e+03   
mean   2003.381174   107.951231     6.444682  9.354221e+04  3.049377e+07   
std      11.587292    18.858011     0.986718  1.606785e+05  3.987824e+07   
min    1927.000000    63.000000     1.400000  1.000000e+02  8.200000e+01   
25%    1999.000000    95.000000     5.900000  1.141400e+04  5.768389e+06   
50%    2006.000000   105.000000     6.500000  3.749300e+04  1.600000e+07   
75%    2012.000000   118.000000     7.100000  1.036910e+05  3.800000e+07   
max    2019.000000   272.000000     9.300000  2.159628e+06  3.560000e+08   

       usa_gross_income  worldwide_gross_income    metascore  \
count      6.459000e+03            6.459000e+03  6459.000000   
mean       4.097111e+07            8.829865e+07    54.955411   
std        6.704754e+07            1.743236e+08    18.092700   
min        9.500000e+01            9.500000e+01     1.000000   
25%        1.548588e+06            4.631308e+06    42.000000   
50%        1.733637e+07            2.685746e+07    55.000000   
75%        5.156309e+07            9.209381e+07    68.000000   
max        9.366622e+08            2.797801e+09   100.000000   

       reviews_from_users  reviews_from_critics  
count         6459.000000           6459.000000  
mean           297.004800            147.188574  
std            451.897507            124.048523  
min              1.000000              1.000000  
25%             74.000000             58.000000  
50%            164.000000            113.000000  
75%            339.000000            197.000000  
max           8302.000000            987.000000

In [6]:
df.loc[df['usa_gross_income'] <= 95]

title  year                     genre  duration       country  \
4104  The Objective  2008  Horror, Sci-Fi, Thriller        90  USA, Morocco   

             language       director                         writer  \
4104  English, Pushto  Daniel Myrick  Daniel Myrick, Mark A. Patton   

     production_company                                             actors  \
4104          Jaz Films  Jonas Ball, Matthew R. Anderson, Jon Huertas, ...   

      avg_vote  votes   budget  usa_gross_income  worldwide_gross_income  \
4104       5.5   7634  4000000                95                      95   

      metascore  reviews_from_users  reviews_from_critics  
4104       26.0                67.0                  37.0

In [7]:
# according to outside research (IMDB, BoxOfficMojo, RottenTomatoes) the gross is correct, however it's a heavy outlier 
# and is more than likely an error, so it will be droped
df.drop(index=4104, inplace=True)
df.reset_index(drop=True, inplace=True)
df.describe()

year     duration     avg_vote         votes        budget  \
count  6458.000000  6458.000000  6458.000000  6.458000e+03  6.458000e+03   
mean   2003.380458   107.954011     6.444828  9.355551e+04  3.049788e+07   
std      11.588047    18.858148     0.986724  1.606874e+05  3.987996e+07   
min    1927.000000    63.000000     1.400000  1.000000e+02  8.200000e+01   
25%    1999.000000    95.000000     5.900000  1.142525e+04  5.777000e+06   
50%    2006.000000   105.000000     6.500000  3.749450e+04  1.600000e+07   
75%    2012.000000   118.000000     7.100000  1.037115e+05  3.800000e+07   
max    2019.000000   272.000000     9.300000  2.159628e+06  3.560000e+08   

       usa_gross_income  worldwide_gross_income    metascore  \
count      6.458000e+03            6.458000e+03  6458.000000   
mean       4.097745e+07            8.831232e+07    54.959895   
std        6.705080e+07            1.743337e+08    18.090512   
min        4.230000e+02            4.230000e+02     1.000000   
25%        1.556678e+06            4.636574e+06    42.000000   
50%        1.733998e+07            2.687465e+07    55.000000   
75%        5.157166e+07            9.213769e+07    68.000000   
max        9.366622e+08            2.797801e+09   100.000000   

       reviews_from_users  reviews_from_critics  
count         6458.000000           6458.000000  
mean           297.040415            147.205636  
std            451.923433            124.050548  
min              1.000000              1.000000  
25%             74.250000             58.000000  
50%            164.000000            113.000000  
75%            339.000000            197.000000  
max           8302.000000            987.000000

In [8]:
df.loc[df['worldwide_gross_income'] <= 500]

title  year             genre  duration country language  \
3164  The Dark Hours  2005  Horror, Thriller        80  Canada  English   

      director    writer production_company  \
3164  Paul Fox  Wil Zmak  Calder Road Films   

                                                 actors  avg_vote  votes  \
3164  Kate Greenhouse, Bruce McFee, Jeff Seymour, Da...       6.0   5165   

      budget  usa_gross_income  worldwide_gross_income  metascore  \
3164  355000               423                     423       63.0   

      reviews_from_users  reviews_from_critics  
3164                51.0                  78.0

In [9]:
# this also appears to be an error so it'll be droped
df.drop(index=3164, inplace=True)
df.reset_index(drop=True, inplace=True)
df.describe()

year     duration     avg_vote         votes        budget  \
count  6457.000000  6457.000000  6457.000000  6.457000e+03  6.457000e+03   
mean   2003.380208   107.958340     6.444897  9.356920e+04  3.050255e+07   
std      11.588927    18.856399     0.986785  1.606961e+05  3.988129e+07   
min    1927.000000    63.000000     1.400000  1.000000e+02  8.200000e+01   
25%    1999.000000    95.000000     5.900000  1.144100e+04  5.777000e+06   
50%    2006.000000   105.000000     6.500000  3.749600e+04  1.600000e+07   
75%    2012.000000   118.000000     7.100000  1.037320e+05  3.800000e+07   
max    2019.000000   272.000000     9.300000  2.159628e+06  3.560000e+08   

       usa_gross_income  worldwide_gross_income    metascore  \
count      6.457000e+03            6.457000e+03  6457.000000   
mean       4.098380e+07            8.832600e+07    54.958650   
std        6.705405e+07            1.743437e+08    18.091636   
min        5.090000e+02            5.940000e+02     1.000000   
25%        1.558836e+06            4.644108e+06    42.000000   
50%        1.734360e+07            2.689185e+07    55.000000   
75%        5.158024e+07            9.218157e+07    68.000000   
max        9.366622e+08            2.797801e+09   100.000000   

       reviews_from_users  reviews_from_critics  
count         6457.000000           6457.000000  
mean           297.078519            147.216354  
std            451.948057            124.057165  
min              1.000000              1.000000  
25%             75.000000             58.000000  
50%            164.000000            113.000000  
75%            339.000000            197.000000  
max           8302.000000            987.000000

In [10]:
df.loc[df['worldwide_gross_income'] <= 1_000]

title  year                    genre  duration  \
181                The Last Wave  1977  Drama, Fantasy, Mystery       106   
2329              The Jimmy Show  2001            Comedy, Drama        96   
3147                 Steve + Sky  2004   Romance, Drama, Comedy        97   
3642  Shake Hands with the Devil  2007      Drama, History, War       112   
5123                  Edwin Boyd  2011             Crime, Drama       105   

        country                      language              director  \
181   Australia  English, Italian, Aboriginal            Peter Weir   
2329        USA                       English          Frank Whaley   
3147    Belgium                         Dutch  Felix van Groeningen   
3642     Canada               English, French    Roger Spottiswoode   
5123     Canada                       English       Nathan Morlando   

                                   writer          production_company  \
181             Peter Weir, Tony Morphett           McElroy & McElroy   
2329  Jonathan Marc Sherman, Frank Whaley  Next Wednesday Productions   
3147                 Felix van Groeningen          Favourite Films NV   
3642      Roméo Dallaire, Michael Donovan     Barna-Alper Productions   
5123                      Nathan Morlando                   IFC Films   

                                                 actors  avg_vote  votes  \
181   Richard Chamberlain, Olivia Hamnett, David Gul...       7.1   7959   
2329  Frank Whaley, Carla Gugino, Ethan Hawke, Lynn ...       5.2    514   
3147  Titus De Voogdt, Delfine Bafort, Johan Heldenb...       6.7   1000   
3642  Roy Dupuis, Owen Sejake, James Gallanders, Mic...       7.6   3084   
5123  Scott Speedman, Kelly Reilly, Kevin Durand, Jo...       6.2   4057   

       budget  usa_gross_income  worldwide_gross_income  metascore  \
181    518400               957                     957       85.0   
2329  1500000              1000                    1000       39.0   
3147   817500               624                     624       44.0   
3642  7810000               594                     594       57.0   
5123  5000000               625                     625       56.0   

      reviews_from_users  reviews_from_critics  
181                 52.0                  68.0  
2329                20.0                   9.0  
3147                 4.0                  12.0  
3642                18.0                  21.0  
5123                20.0                  33.0

In [11]:
# "The Last Wave" worldwide gross is for a release and the original box office gross is not know, so drop it
# https://www.boxofficemojo.com/title/tt0076299/?ref_=bo_se_r_1

# due to inconsistancies "The Jimmy Show" will also be droped
# the rest are fine

df.drop(index=[181, 2329], inplace=True)
df.reset_index(drop=True, inplace=True)
df.describe()

year     duration     avg_vote         votes        budget  \
count  6455.000000  6455.000000  6455.000000  6.455000e+03  6.455000e+03   
mean   2003.384663   107.960496     6.444988  9.359688e+04  3.051168e+07   
std      11.586031    18.858717     0.986782  1.607133e+05  3.988408e+07   
min    1927.000000    63.000000     1.400000  1.000000e+02  8.200000e+01   
25%    1999.000000    95.000000     5.900000  1.146200e+04  5.788500e+06   
50%    2006.000000   105.000000     6.500000  3.756400e+04  1.600000e+07   
75%    2012.000000   118.000000     7.100000  1.037355e+05  3.800000e+07   
max    2019.000000   272.000000     9.300000  2.159628e+06  3.560000e+08   

       usa_gross_income  worldwide_gross_income    metascore  \
count      6.455000e+03            6.455000e+03  6455.000000   
mean       4.099650e+07            8.835337e+07    54.956468   
std        6.706056e+07            1.743638e+08    18.089484   
min        5.090000e+02            5.940000e+02     1.000000   
25%        1.568678e+06            4.644498e+06    42.000000   
50%        1.736802e+07            2.689548e+07    55.000000   
75%        5.159012e+07            9.223271e+07    68.000000   
max        9.366622e+08            2.797801e+09   100.000000   

       reviews_from_users  reviews_from_critics  
count         6455.000000           6455.000000  
mean           297.159411            147.250039  
std            451.994617            124.060533  
min              1.000000              1.000000  
25%             75.000000             58.000000  
50%            164.000000            113.000000  
75%            339.500000            197.000000  
max           8302.000000            987.000000

In [12]:
# save our changes
df.to_csv('../../Data/Movies/IMDB_Movies_cleaned_v02.csv', index=False)

___
## production_company dummy

In [13]:
df['production_company'].value_counts()

Universal Pictures             296
Warner Bros.                   291
Columbia Pictures              284
Paramount Pictures             273
Twentieth Century Fox          207
                              ... 
Django Film                      1
Double Nickel Entertainment      1
Les Cinémas de la Zone           1
Margin Films                     1
Chartoff Productions             1
Name: production_company, Length: 2384, dtype: int64

Since there are 2384 different production companies, dummying out each one would be too much so I'd like to combine some of the lesser occuring ones into their own category.

In [14]:
# make a list of production companies that appear more than 4 times in our data set
prod_cos = []

for i, prod_co_count in enumerate(df['production_company'].value_counts()):
    if prod_co_count >= 5:
        prod_cos.append(df['production_company'].value_counts().index[i])

# next we need to update our data frame so that each production company with less than 10 entries gets catagorized
# as 'other'
df['production_company'] = df['production_company'].map(lambda x: x if x in prod_cos else 'other').values

In [15]:
# creating dummies for production_company
pc_dummy_df = pd.get_dummies(df['production_company']).add_prefix('prod_co-')
pc_dummy_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace= True)
pc_dummy_df.drop('prod_co-other', axis=1, inplace=True)
pc_dummy_df.head()

prod_co-2929_productions  prod_co-3_arts_entertainment  \
0                         0                             0   
1                         0                             0   
2                         0                             0   
3                         0                             0   
4                         0                             0   

   prod_co-40_acres_&_a_mule_filmworks  prod_co-aardman_animations  \
0                                    0                           0   
1                                    0                           0   
2                                    0                           0   
3                                    0                           0   
4                                    0                           0   

   prod_co-affirm_films  prod_co-alcon_entertainment  \
0                     0                            0   
1                     0                            0   
2                     0                            0   
3                     0                            0   
4                     0                            0   

   prod_co-alliance_atlantis_communications  prod_co-amblin_entertainment  \
0                                         0                             0   
1                                         0                             0   
2                                         0                             0   
3                                         0                             0   
4                                         0                             0   

   prod_co-american_zoetrope  prod_co-annapurna_pictures  ...  \
0                          0                           0  ...   
1                          0                           0  ...   
2                          0                           0  ...   
3                          0                           0  ...   
4                          0                           0  ...   

   prod_co-walt_disney_feature_animation  prod_co-walt_disney_pictures  \
0                                      0                             0   
1                                      0                             0   
2                                      0                             0   
3                                      0                             0   
4                                      0                             0   

   prod_co-walt_disney_productions  prod_co-warner_bros.  \
0                                0                     0   
1                                0                     0   
2                                0                     0   
3                                1                     0   
4                                0                     0   

   prod_co-warner_independent_pictures_(wip)  prod_co-why_not_productions  \
0                                          0                            0   
1                                          0                            0   
2                                          0                            0   
3                                          0                            0   
4                                          0                            0   

   prod_co-working_title_films  prod_co-worldview_entertainment  \
0                            0                                0   
1                            0                                0   
2                            0                                0   
3                            0                                0   
4                            0                                0   

   prod_co-zentropa_entertainments  prod_co-zoetrope_studios  
0                                0                         0  
1                                0                         0  
2                                0                         0  
3                                0                         0  
4                                0   

___
## genre dummy

In [16]:
genre_dummy_df = df['genre'].map(lambda x: x.split(', ')).str.join('|').str.get_dummies().add_prefix('genre-')
genre_dummy_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace=True)
genre_dummy_df.head()

genre-action  genre-adventure  genre-animation  genre-biography  \
0             0                0                0                0   
1             0                0                0                0   
2             0                0                0                0   
3             0                0                1                0   
4             0                0                0                0   

   genre-comedy  genre-crime  genre-drama  genre-family  genre-fantasy  \
0             0            0            1             0              0   
1             1            0            1             0              0   
2             1            0            1             1              0   
3             0            0            0             1              1   
4             0            0            1             0              0   

   genre-film-noir  ...  genre-horror  genre-music  genre-musical  \
0                0  ...             0            0              0   
1                0  ...             0            0              0   
2                0  ...             0            0              0   
3                0  ...             0            0              0   
4                0  ...             0            0              0   

   genre-mystery  genre-romance  genre-sci-fi  genre-sport  genre-thriller  \
0              0              0             1            0               0   
1              0              1             0            0               0   
2              0              0             0            0               0   
3              0              0             0            0               0   
4              0              1             0            0               0   

   genre-war  genre-western  
0          0              0  
1          0              0  
2          0              0  
3          0              0  
4          0              0  

[5 rows x 21 columns]

___
## country dummy

In [17]:
country_dum_df = df['country'].map(lambda x: x.split(', ')).str.join('|').str.get_dummies().add_prefix('country-')
country_dum_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace = True)
country_dum_df.head()

country-afghanistan  country-algeria  country-angola  country-argentina  \
0                    0                0               0                  0   
1                    0                0               0                  0   
2                    0                0               0                  0   
3                    0                0               0                  0   
4                    0                0               0                  0   

   country-aruba  country-australia  country-austria  country-bahamas  \
0              0                  0                0                0   
1              0                  0                0                0   
2              0                  0                0                0   
3              0                  0                0                0   
4              0                  0                0                0   

   country-belgium  country-bhutan  ...  country-turkey  country-uk  \
0                0               0  ...               0           0   
1                0               0  ...               0           0   
2                0               0  ...               0           0   
3                0               0  ...               0           0   
4                0               0  ...               0           0   

   country-usa  country-ukraine  country-united_arab_emirates  \
0            0                0                             0   
1            1                0                             0   
2            1                0                             0   
3            1                0                             0   
4            1                0                             0   

   country-uruguay  country-venezuela  country-vietnam  country-west_germany  \
0                0                  0                0                     0   
1                0                  0                0                     0   
2                0                  0                0                     0   
3                0                  0                0                     0   
4                0                  0                0                     0   

   country-yugoslavia  
0                   0  
1                   0  
2                   0  
3                   0  
4                   0  

[5 rows x 110 columns]

___
## writer dummy

In [18]:
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language                      director  \
0       153  Germany   German                    Fritz Lang   
1        87      USA  English               Charles Chaplin   
2        87      USA  English               Charles Chaplin   
3        83      USA  English  William Cottrell, David Hand   
4       238      USA  English  Victor Fleming, George Cukor   

                             writer       production_company  \
0  Thea von Harbou, Thea von Harbou                    other   
1                   Charles Chaplin                    other   
2                   Charles Chaplin                    other   
3        Jacob Grimm, Wilhelm Grimm  Walt Disney Productions   
4  Margaret Mitchell, Sidney Howard                    other   

                                              actors  avg_vote   votes  \
0  Alfred Abel, Gustav Fröhlich, Rudolf Klein-Rog...       8.3  148396   
1  Virginia Cherrill, Florence Lee, Harry Myers, ...       8.5  152716   
2  Charles Chaplin, Paulette Goddard, Henry Bergm...       8.5  197969   
3  Roy Atwell, Stuart Buchanan, Adriana Caselotti...       7.6  168735   
4  Thomas Mitchell, Barbara O'Neil, Vivien Leigh,...       8.1  269664   

    budget  usa_gross_income  worldwide_gross_income  metascore  \
0   780000           1236166                 1349711       98.0   
1  1500000             19181                   32609       99.0   
2  1500000            163577                  445226       96.0   
3  1499000         184925486               184925486       95.0   
4  3977000         200852579               402352579       97.0   

   reviews_from_users  reviews_from_critics  
0               471.0                 194.0  
1               270.0                 120.0  
2               262.0                 146.0  
3               229.0                 166.0  
4               808.0                 183.0

In [19]:
# Since the writer column can have multiple entries per movie, we'll have to approach this a little differently

# this line of code converts each writer entry into a series then counts the total
writer_counts = df['writer'].str.split(', ').apply(lambda x: pd.Series(x).value_counts()).sum()
writer_counts.sort_values(ascending=False).head(60)

Stephen King             38.0
Woody Allen              33.0
Luc Besson               28.0
Tyler Perry              22.0
Ethan Coen               20.0
David Koepp              19.0
Joel Coen                18.0
Sylvester Stallone       17.0
John Carpenter           17.0
John Hughes              17.0
Brian Helgeland          15.0
Quentin Tarantino        15.0
Akiva Goldsman           14.0
Robert Mark Kamen        14.0
Richard Maibaum          14.0
Wes Craven               14.0
Hayao Miyazaki           13.0
George Lucas             13.0
Michael Crichton         13.0
M. Night Shyamalan       13.0
David Mamet              13.0
Robert Rodriguez         13.0
Zak Penn                 12.0
Ehren Kruger             12.0
John Logan               12.0
Francis Ford Coppola     12.0
Leigh Whannell           11.0
Oliver Stone             11.0
Christopher McQuarrie    11.0
Steven Knight            11.0
George A. Romero         11.0
Mark Bomback             11.0
Guillermo del Toro       11.0
James Came

In [20]:
# now we can use a similar method we used for production company
top_writers = []

for i, writer_count in enumerate(writer_counts):
    if writer_count >= 8:
        top_writers.append(writer_counts.index[i])
        
len(top_writers)

118

In [21]:
updated_writers = []

for entry in df['writer']:
    
    writers = entry.split(', ')
    
    new_entry = []
    
    for writer in writers:
        
        if writer in top_writers:
            new_entry.append(writer)
        else:
            new_entry.append('other')
            
    updated_writers.append(new_entry)

len(updated_writers)

6455

In [22]:
df.shape

(6455, 18)

In [23]:
df['writer'] = [', '.join(x) for x in updated_writers]
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language                      director        writer  \
0       153  Germany   German                    Fritz Lang  other, other   
1        87      USA  English               Charles Chaplin         other   
2        87      USA  English               Charles Chaplin         other   
3        83      USA  English  William Cottrell, David Hand  other, other   
4       238      USA  English  Victor Fleming, George Cukor  other, other   

        production_company                                             actors  \
0                    other  Alfred Abel, Gustav Fröhlich, Rudolf Klein-Rog...   
1                    other  Virginia Cherrill, Florence Lee, Harry Myers, ...   
2                    other  Charles Chaplin, Paulette Goddard, Henry Bergm...   
3  Walt Disney Productions  Roy Atwell, Stuart Buchanan, Adriana Caselotti...   
4                    other  Thomas Mitchell, Barbara O'Neil, Vivien Leigh,...   

   avg_vote   votes   budget  usa_gross_income  worldwide_gross_income  \
0       8.3  148396   780000           1236166                 1349711   
1       8.5  152716  1500000             19181                   32609   
2       8.5  197969  1500000            163577                  445226   
3       7.6  168735  1499000         184925486               184925486   
4       8.1  269664  3977000         200852579               402352579   

   metascore  reviews_from_users  reviews_from_critics  
0       98.0               471.0                 194.0  
1       99.0               270.0                 120.0  
2       96.0               262.0                 146.0  
3       95.0               229.0                 166.0  
4       97.0               808.0                 183.0

In [24]:
writer_dum_df = df['writer'].map(lambda x: x.split(', ')).str.join('|').str.get_dummies().add_prefix('writer-')
writer_dum_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace = True)
writer_dum_df.drop('writer-other', axis=1, inplace=True)
writer_dum_df.head()

writer-aaron_sorkin  writer-adam_sandler  writer-akiva_goldsman  \
0                    0                    0                      0   
1                    0                    0                      0   
2                    0                    0                      0   
3                    0                    0                      0   
4                    0                    0                      0   

   writer-alex_kurtzman  writer-aline_brosh_mckenna  writer-allan_loeb  \
0                     0                           0                  0   
1                     0                           0                  0   
2                     0                           0                  0   
3                     0                           0                  0   
4                     0                           0                  0   

   writer-andrew_stanton  writer-billy_ray  writer-brian_helgeland  \
0                      0                 0                       0   
1                      0                 0                       0   
2                      0                 0                       0   
3                      0                 0                       0   
4                      0                 0                       0   

   writer-charles_leavitt  ...  writer-ted_elliott  writer-terry_gilliam  \
0                       0  ...                   0                     0   
1                       0  ...                   0                     0   
2                       0  ...                   0                     0   
3                       0  ...                   0                     0   
4                       0  ...                   0                     0   

   writer-tony_gilroy  writer-tyler_perry  writer-wes_anderson  \
0                   0                   0                    0   
1                   0                   0                    0   
2                   0                   0                    0   
3                   0                   0                    0   
4                   0                   0                    0   

   writer-wes_craven  writer-william_goldman  writer-william_shakespeare  \
0                  0                       0                           0   
1                  0                       0                           0   
2                  0                       0                           0   
3                  0                       0                           0   
4                  0                       0                           0   

   writer-woody_allen  writer-zak_penn  
0                   0                0  
1                   0                0  
2                   0                0  
3                   0                0  
4                   0                0  

[5 rows x 118 columns]

___
## language dummy

In [25]:
lang_dum_df = df['language'].map(lambda x: x.replace(', ', '|')).str.get_dummies().add_prefix('language-')
lang_dum_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace = True)
lang_dum_df.head()

language-aboriginal  language-acholi  language-afrikaans  language-akan  \
0                    0                0                   0              0   
1                    0                0                   0              0   
2                    0                0                   0              0   
3                    0                0                   0              0   
4                    0                0                   0              0   

   language-albanian  language-algonquin  language-american_sign_language  \
0                  0                   0                                0   
1                  0                   0                                0   
2                  0                   0                                0   
3                  0                   0                                0   
4                  0                   0                                0   

   language-amharic  language-ancient_(to_1453)  language-apache_languages  \
0                 0                           0                          0   
1                 0                           0                          0   
2                 0                           0                          0   
3                 0                           0                          0   
4                 0                           0                          0   

   ...  language-urdu  language-uzbek  language-vietnamese  language-visayan  \
0  ...              0               0                    0                 0   
1  ...              0               0                    0                 0   
2  ...              0               0                    0                 0   
3  ...              0               0                    0                 0   
4  ...              0               0                    0                 0   

   language-washoe  language-welsh  language-xhosa  language-yiddish  \
0                0               0               0                 0   
1                0               0               0                 0   
2                0               0               0                 0   
3                0               0               0                 0   
4                0               0               0                 0   

   language-yoruba  language-zulu  
0                0              0  
1                0              0  
2                0              0  
3                0              0  
4                0              0  

[5 rows x 160 columns]

___
## director dummy

In [26]:
df['director'].value_counts()

Clint Eastwood                      34
Woody Allen                         33
Steven Spielberg                    31
Steven Soderbergh                   25
Ridley Scott                        24
                                    ..
Dallas Jenkins                       1
Michael Haigney, Kunihiko Yuyama     1
Steve Barron                         1
Xavier Giannoli                      1
Frank Miller                         1
Name: director, Length: 2917, dtype: int64

In [27]:
# same method as writer

# this line of code converts each writer entry into a series then counts the total
dir_counts = df['director'].str.split(', ').apply(lambda x: pd.Series(x).value_counts()).sum()
dir_counts.sort_values(ascending=False).head(60)

Clint Eastwood          34.0
Woody Allen             33.0
Steven Spielberg        31.0
Steven Soderbergh       25.0
Ridley Scott            24.0
Ron Howard              22.0
Martin Scorsese         22.0
Tim Burton              19.0
Robert Zemeckis         18.0
Oliver Stone            17.0
Robert Rodriguez        17.0
Ethan Coen              17.0
Joel Coen               17.0
Francis Ford Coppola    17.0
Brian De Palma          17.0
Spike Lee               17.0
Renny Harlin            16.0
Rob Reiner              16.0
Joel Schumacher         16.0
Barry Levinson          16.0
Tyler Perry             15.0
Richard Linklater       15.0
Richard Donner          15.0
John Carpenter          15.0
David Cronenberg        14.0
Tony Scott              14.0
Sam Raimi               14.0
Ivan Reitman            14.0
Wes Craven              14.0
Stephen Frears          13.0
Roland Emmerich         13.0
Mike Nichols            13.0
Peter Farrelly          13.0
Michael Bay             13.0
Bobby Farrelly

In [28]:
# now we can us a similar method we used for production company
top_dirs = []

for i, dir_count in enumerate(dir_counts):
    if dir_count >= 8:
        top_dirs.append(dir_counts.index[i])
        

updated_dirs = []

for entry in df['director']:
    
    dirs = entry.split(', ')
    
    new_entry = []
    
    for director in dirs:
        
        if director in top_dirs:
            new_entry.append(director)
        else:
            new_entry.append('other')
            
    updated_dirs.append(new_entry)

len(updated_dirs)

6455

In [29]:
len(top_dirs)

132

In [30]:
df.shape

(6455, 18)

In [31]:
df['director'] = [', '.join(x) for x in updated_dirs]
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language      director        writer  \
0       153  Germany   German         other  other, other   
1        87      USA  English         other         other   
2        87      USA  English         other         other   
3        83      USA  English  other, other  other, other   
4       238      USA  English  other, other  other, other   

        production_company                                             actors  \
0                    other  Alfred Abel, Gustav Fröhlich, Rudolf Klein-Rog...   
1                    other  Virginia Cherrill, Florence Lee, Harry Myers, ...   
2                    other  Charles Chaplin, Paulette Goddard, Henry Bergm...   
3  Walt Disney Productions  Roy Atwell, Stuart Buchanan, Adriana Caselotti...   
4                    other  Thomas Mitchell, Barbara O'Neil, Vivien Leigh,...   

   avg_vote   votes   budget  usa_gross_income  worldwide_gross_income  \
0       8.3  148396   780000           1236166                 1349711   
1       8.5  152716  1500000             19181                   32609   
2       8.5  197969  1500000            163577                  445226   
3       7.6  168735  1499000         184925486               184925486   
4       8.1  269664  3977000         200852579               402352579   

   metascore  reviews_from_users  reviews_from_critics  
0       98.0               471.0                 194.0  
1       99.0               270.0                 120.0  
2       96.0               262.0                 146.0  
3       95.0               229.0                 166.0  
4       97.0               808.0                 183.0

In [32]:
dir_dum_df = df['director'].map(lambda x: x.replace(', ', '|')).str.get_dummies().add_prefix('director-')
dir_dum_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace=True)
dir_dum_df.drop('director-other', axis=1, inplace=True)
dir_dum_df.head()

director-adam_shankman  director-alan_parker  director-andrew_davis  \
0                       0                     0                      0   
1                       0                     0                      0   
2                       0                     0                      0   
3                       0                     0                      0   
4                       0                     0                      0   

   director-ang_lee  director-antoine_fuqua  director-barry_levinson  \
0                 0                       0                        0   
1                 0                       0                        0   
2                 0                       0                        0   
3                 0                       0                        0   
4                 0                       0                        0   

   director-barry_sonnenfeld  director-bill_condon  director-bobby_farrelly  \
0                          0                     0                        0   
1                          0                     0                        0   
2                          0                     0                        0   
3                          0                     0                        0   
4                          0                     0                        0   

   director-brett_ratner  ...  director-tyler_perry  director-walter_hill  \
0                      0  ...                     0                     0   
1                      0  ...                     0                     0   
2                      0  ...                     0                     0   
3                      0  ...                     0                     0   
4                      0  ...                     0                     0   

   director-wayne_wang  director-wes_anderson  director-wes_craven  \
0                    0                      0                    0   
1                    0                      0                    0   
2                    0                      0                    0   
3                    0                      0                    0   
4                    0                      0                    0   

   director-william_friedkin  director-wolfgang_petersen  \
0                          0                           0   
1                          0                           0   
2                          0                           0   
3                          0                           0   
4                          0                           0   

   director-woody_allen  director-yimou_zhang  director-zack_snyder  
0                     0                     0                     0  
1                     0                     0                     0  
2                     0                     0                     0  
3                     0                     0                     0  
4                     0                     0                     0  

[5 rows x 132 columns]

___
## actors dummy

In [33]:
# same method as director

# this line of code converts each writer entry into a series then counts the total
act_counts = df['actors'].str.split(', ').apply(lambda x: pd.Series(x).value_counts()).sum()
act_counts.sort_values(ascending=False).head(60)

Samuel L. Jackson      69.0
Robert De Niro         66.0
Morgan Freeman         55.0
Bruce Willis           51.0
Nicolas Cage           50.0
Steve Buscemi          47.0
Woody Harrelson        47.0
Liam Neeson            46.0
Alec Baldwin           46.0
Richard Jenkins        45.0
Robert Downey Jr.      44.0
J.K. Simmons           43.0
Willem Dafoe           43.0
Julianne Moore         43.0
Johnny Depp            42.0
Owen Wilson            42.0
Matt Damon             42.0
Christopher Walken     42.0
Tom Hanks              42.0
John Goodman           42.0
Brad Pitt              41.0
Danny DeVito           41.0
Bill Murray            41.0
Harrison Ford          41.0
Nicole Kidman          40.0
Mark Wahlberg          40.0
Susan Sarandon         39.0
Robin Williams         39.0
Stanley Tucci          39.0
Ben Kingsley           39.0
Tom Cruise             38.0
Michael Caine          38.0
Clint Eastwood         38.0
Keith David            38.0
Cate Blanchett         38.0
Matthew McConaughey 

In [34]:
# now we can us a similar method we used for production company
top_actors = []

for i, act_count in enumerate(act_counts):
    if act_count >= 20:
        top_actors.append(act_counts.index[i])
        
print(len(top_actors))
        
updated_acts = []

for entry in df['actors']:
    
    actors = entry.split(', ')
    
    new_entry = []
    
    for actor in actors:
        
        if actor in top_actors:
            new_entry.append(actor)
        else:
            new_entry.append('other')
            
    updated_acts.append(new_entry)

len(updated_acts)

410


6455

In [35]:
df['actors'] = [', '.join(x) for x in updated_acts]
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language      director        writer  \
0       153  Germany   German         other  other, other   
1        87      USA  English         other         other   
2        87      USA  English         other         other   
3        83      USA  English  other, other  other, other   
4       238      USA  English  other, other  other, other   

        production_company                                             actors  \
0                    other  other, other, other, other, other, other, othe...   
1                    other           other, other, other, other, other, other   
2                    other  other, other, other, other, other, other, othe...   
3  Walt Disney Productions  other, other, other, other, other, other, othe...   
4                    other  other, other, other, other, other, other, othe...   

   avg_vote   votes   budget  usa_gross_income  worldwide_gross_income  \
0       8.3  148396   780000           1236166                 1349711   
1       8.5  152716  1500000             19181                   32609   
2       8.5  197969  1500000            163577                  445226   
3       7.6  168735  1499000         184925486               184925486   
4       8.1  269664  3977000         200852579               402352579   

   metascore  reviews_from_users  reviews_from_critics  
0       98.0               471.0                 194.0  
1       99.0               270.0                 120.0  
2       96.0               262.0                 146.0  
3       95.0               229.0                 166.0  
4       97.0               808.0                 183.0

In [36]:
actor_dum_df = df['actors'].map(lambda x: x.replace(', ', '|')).str.get_dummies().add_prefix('actor-')
actor_dum_df.rename(columns = lambda x: x.replace(' ', '_').lower(), inplace=True)
actor_dum_df.drop('actor-other', axis=1, inplace=True)
actor_dum_df.head()

actor-aaron_eckhart  actor-adam_sandler  actor-adrien_brody  \
0                    0                   0                   0   
1                    0                   0                   0   
2                    0                   0                   0   
3                    0                   0                   0   
4                    0                   0                   0   

   actor-al_pacino  actor-alan_arkin  actor-alan_tudyk  actor-alec_baldwin  \
0                0                 0                 0                   0   
1                0                 0                 0                   0   
2                0                 0                 0                   0   
3                0                 0                 0                   0   
4                0                 0                 0                   0   

   actor-alfred_molina  actor-allison_janney  actor-amanda_peet  ...  \
0                    0                     0                  0  ...   
1                    0                     0                  0  ...   
2                    0                     0                  0  ...   
3                    0                     0                  0  ...   
4                    0                     0                  0  ...   

   actor-william_fichtner  actor-william_h._macy  actor-william_hurt  \
0                       0                      0                   0   
1                       0                      0                   0   
2                       0                      0                   0   
3                       0                      0                   0   
4                       0                      0                   0   

   actor-winona_ryder  actor-woody_harrelson  actor-xander_berkeley  \
0                   0                      0                      0   
1                   0                      0                      0   
2                   0                      0                      0   
3                   0                      0                      0   
4                   0                      0                      0   

   actor-zac_efron  actor-zach_galifianakis  actor-zoe_saldana  \
0                0                        0                  0   
1                0                        0                  0   
2                0                        0                  0   
3                0                        0                  0   
4                0                        0                  0   

   actor-zooey_deschanel  
0                      0  
1                      0  
2                      0  
3                      0  
4                      0  

[5 rows x 410 columns]

___
## Combining our dummy data to the main df

In [37]:
df.head()

title  year                       genre  \
0                       Metropolis  1927               Drama, Sci-Fi   
1                      City Lights  1931      Comedy, Drama, Romance   
2                     Modern Times  1936       Comedy, Drama, Family   
3  Snow White and the Seven Dwarfs  1937  Animation, Family, Fantasy   
4               Gone with the Wind  1939     Drama, History, Romance   

   duration  country language      director        writer  \
0       153  Germany   German         other  other, other   
1        87      USA  English         other         other   
2        87      USA  English         other         other   
3        83      USA  English  other, other  other, other   
4       238      USA  English  other, other  other, other   

        production_company                                             actors  \
0                    other  other, other, other, other, other, other, othe...   
1                    other           other, other, other, other, other, other   
2                    other  other, other, other, other, other, other, othe...   
3  Walt Disney Productions  other, other, other, other, other, other, othe...   
4                    other  other, other, other, other, other, other, othe...   

   avg_vote   votes   budget  usa_gross_income  worldwide_gross_income  \
0       8.3  148396   780000           1236166                 1349711   
1       8.5  152716  1500000             19181                   32609   
2       8.5  197969  1500000            163577                  445226   
3       7.6  168735  1499000         184925486               184925486   
4       8.1  269664  3977000         200852579               402352579   

   metascore  reviews_from_users  reviews_from_critics  
0       98.0               471.0                 194.0  
1       99.0               270.0                 120.0  
2       96.0               262.0                 146.0  
3       95.0               229.0                 166.0  
4       97.0               808.0                 183.0

In [38]:
df_dummied = pd.concat([
    df,
    country_dum_df,
    lang_dum_df,
    dir_dum_df,
    writer_dum_df,
    genre_dummy_df,
    pc_dummy_df,
    actor_dum_df,
    
], axis=1).drop(columns=['genre', 'language', 'country', 'director', 'writer', 'production_company', 'actors'])

df_dummied.dtypes

title                       object
year                         int64
duration                     int64
avg_vote                   float64
votes                        int64
                            ...   
actor-xander_berkeley        int64
actor-zac_efron              int64
actor-zach_galifianakis      int64
actor-zoe_saldana            int64
actor-zooey_deschanel        int64
Length: 1104, dtype: object

In [39]:
df_dummied.head()

title  year  duration  avg_vote   votes   budget  \
0                       Metropolis  1927       153       8.3  148396   780000   
1                      City Lights  1931        87       8.5  152716  1500000   
2                     Modern Times  1936        87       8.5  197969  1500000   
3  Snow White and the Seven Dwarfs  1937        83       7.6  168735  1499000   
4               Gone with the Wind  1939       238       8.1  269664  3977000   

   usa_gross_income  worldwide_gross_income  metascore  reviews_from_users  \
0           1236166                 1349711       98.0               471.0   
1             19181                   32609       99.0               270.0   
2            163577                  445226       96.0               262.0   
3         184925486               184925486       95.0               229.0   
4         200852579               402352579       97.0               808.0   

   ...  actor-william_fichtner  actor-william_h._macy  actor-william_hurt  \
0  ...                       0                      0                   0   
1  ...                       0                      0                   0   
2  ...                       0                      0                   0   
3  ...                       0                      0                   0   
4  ...                       0                      0                   0   

   actor-winona_ryder  actor-woody_harrelson  actor-xander_berkeley  \
0                   0                      0                      0   
1                   0                      0                      0   
2                   0                      0                      0   
3                   0                      0                      0   
4                   0                      0                      0   

   actor-zac_efron  actor-zach_galifianakis  actor-zoe_saldana  \
0                0                        0                  0   
1                0                        0                  0   
2                0                        0                  0   
3                0                        0                  0   
4                0                        0                  0   

   actor-zooey_deschanel  
0                      0  
1                      0  
2                      0  
3                      0  
4                      0  

[5 rows x 1104 columns]

In [40]:
# create a sparse matrix to save our dummied data.
sparse_dummy = csr_matrix(df_dummied.loc[:, 'year':].values)

In [41]:
# save our sparse matrix
save_npz('../../Data/Movies/sparse_dummy.npz', sparse_dummy)

In [42]:
# save our columns as a csv file
np.savetxt(fname='../../Data/Movies/dummy_cols.csv', # filename
           X=df_dummied.columns.values[1:], # data to save
           fmt='%s', # %s means the format is string
           delimiter=",")

In [43]:
np.loadtxt('../../Data/Movies/dummy_cols.csv', dtype=str)

array(['year', 'duration', 'avg_vote', ..., 'actor-zach_galifianakis',
       'actor-zoe_saldana', 'actor-zooey_deschanel'], dtype='<U51')

In [44]:
# Sanity check
pd.DataFrame.sparse.from_spmatrix(load_npz('../../Data/Movies/sparse_dummy.npz'),
                                  index = df_dummied['title'].values,
                                  columns = np.loadtxt('../../Data/Movies/dummy_cols.csv', dtype=str)).head()

year  duration  avg_vote     votes  \
Metropolis                       1927.0     153.0       8.3  148396.0   
City Lights                      1931.0      87.0       8.5  152716.0   
Modern Times                     1936.0      87.0       8.5  197969.0   
Snow White and the Seven Dwarfs  1937.0      83.0       7.6  168735.0   
Gone with the Wind               1939.0     238.0       8.1  269664.0   

                                    budget  usa_gross_income  \
Metropolis                        780000.0         1236166.0   
City Lights                      1500000.0           19181.0   
Modern Times                     1500000.0          163577.0   
Snow White and the Seven Dwarfs  1499000.0       184925486.0   
Gone with the Wind               3977000.0       200852579.0   

                                 worldwide_gross_income  metascore  \
Metropolis                                    1349711.0       98.0   
City Lights                                     32609.0       99.0   
Modern Times                                   445226.0       96.0   
Snow White and the Seven Dwarfs             184925486.0       95.0   
Gone with the Wind                          402352579.0       97.0   

                                 reviews_from_users  reviews_from_critics  \
Metropolis                                    471.0                 194.0   
City Lights                                   270.0                 120.0   
Modern Times                                  262.0                 146.0   
Snow White and the Seven Dwarfs               229.0                 166.0   
Gone with the Wind                            808.0                 183.0   

                                 ...  actor-william_fichtner  \
Metropolis                       ...                     0.0   
City Lights                      ...                     0.0   
Modern Times                     ...                     0.0   
Snow White and the Seven Dwarfs  ...                     0.0   
Gone with the Wind               ...                     0.0   

                                 actor-william_h._macy  actor-william_hurt  \
Metropolis                                         0.0                 0.0   
City Lights                                        0.0                 0.0   
Modern Times                                       0.0                 0.0   
Snow White and the Seven Dwarfs                    0.0                 0.0   
Gone with the Wind                                 0.0                 0.0   

                                 actor-winona_ryder  actor-woody_harrelson  \
Metropolis                                      0.0                    0.0   
City Lights                                     0.0                    0.0   
Modern Times                                    0.0                    0.0   
Snow White and the Seven Dwarfs                 0.0                    0.0   
Gone with the Wind                              0.0                    0.0   

                                 actor-xander_berkeley  actor-zac_efron  \
Metropolis                                         0.0              0.0   
City Lights                                        0.0              0.0   
Modern Times                                       0.0              0.0   
Snow White and the Seven Dwarfs                    0.0              0.0   
Gone with the Wind                                 0.0              0.0   

                                 actor-zach_galifianakis  actor-zoe_saldana  \
Metropolis                                           0.0                0.0   
City Lights                                          0.0                0.0   
Modern Times                                         0.0                0.0   
Snow White and the Seven Dwarfs                      0.0                0.0   
Gone with the Wind                                   0.0                0.0   

                                 actor-zooey_deschanel  
Metropolis           